Engineroom/engineroom.cpp

```diff
@@ -0,0 +1,135 @@
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <cstdlib>
+#include <ctime>
+
+using namespace std;
+
+// Function to read text from a file and append a newline
+string readFromFile(const string& filename)
+{
+    ifstream file(filename);
+    if (!file)
+    {
+        cerr << "Error: Could not open file " << filename << endl;
+        return "";
+    }
+    string content;
+    getline(file, content, '\0'); // Read entire file content into string
+    file.close();
+    content += "\n"; // Append a newline to the content
+    return content;
+}
+
+// Class for the Engine Room
+class EngineRoom
+{
+public:
+    // Function to handle entering the engine room
+    void enter()

```

Comment:
here would need enter(Player &player) if we give the player class the engine piece variable. I added a note in the pull request, but take a look at the conventions branch in github and see what you think!